src/components/Dialogs/Conditions/OrgUnitCondition.js

```diff
@@ -72,6 +79,111 @@ const OrgUnitCondition = ({
         />
     )
 }
+const getMetaDataId = (id) =>
+    ouIdHelper.hasLevelPrefix(id) || ouIdHelper.hasGroupPrefix(id)
+        ? ouIdHelper.removePrefix(id)
+        : id
+
+// VERSION-TOGGLE: rename to OrgUnitCondition when 42 is lowest supported version
+const OrgUnitConditionMinVersionV42 = ({
+    addMetadata,
+    addParentGraphMap,
+    condition,
+    onChange,
+    metadata,
+    parentGraphMap,
+}) => {
+    const { rootOrgUnits, currentUser } = useCachedDataQuery()
+    const selected = useMemo(() => {
+        const idsString = condition.split(':')?.[1] ?? ''
```

Comment:
It seems like we should have a function that does this, somewhere in LL, but since I'm not sure where, it is something we can improve later.

Reply:
Good find, but it turns out I can't use this `parseCondition` helper. The reason is related to my previous [comment here](https://github.com/dhis2/line-listing-app/pull/651#discussion_r2000593435). If `condition` is an empty string, then something unexpected happens: `parseCondtion('')` returns `[ '' ]` and we then the map callback fails.

I suppose I could look into fixing `parseCondition`, because `''` should return `[]` and not `[ '' ]`. But I guess this could have some unintended consequences.

What do you think?

Reply:
yeah, let's just wait on it. Is there a way to flag this with a // REFACTOR or something without triggering a new code smell?